### PR TITLE
ci: 🚭 fix summonerd smoke tests

### LINKED ISSUE #4321: Planner changes broke summoner
**Describe the bug**

I think I broke the summoner by changing the memo API in the planner and not patching it correctly https://github.com/penumbra-zone/penumbra/actions/runs/8960044589/job/24606104297

I think this is probably not setting the contribution address

### LINKED ISSUE #4370: ci: 🧟 summonerd smoke test leaves dangling processes when run locally
> noting for future reference: i found when running this script locally that the `trap` commands herein weren't working as expected; `cometbft` and `pd` were sometimes left running, but i haven't tracked down what causes the `kill` call to fail. :raised_eyebrow:

_originally posted by @cratelyn in https://github.com/penumbra-zone/penumbra/pull/4358#discussion_r1596030768_

### PR DESCRIPTION
**Pull request by cratelyn**
#### describe your changes

this provides some small changes needed for running the summoner smoke test locally.

#### issue ticket number and link

fixes #4321.

fixes #4370.

#### checklist before requesting a review

- [x] if this code contains consensus-breaking changes, i have added the "consensus-breaking" label. otherwise, i declare my belief that there are not consensus-breaking changes, for the following reason:

  > this branch only changes testing infrastructure and development tooling.

### CODE REVIEW

File: .github/workflows/summoner_smoke.yml
Comment (cratelyn) [inline .github/workflows/summoner_smoke.yml:6]: :exclamation: don't merge with this change; i'm interested in temporarily enabling this to show that this _does_ fix the issue. once we've got a fix in hand, we'll squash down these commits and revert this.

File: deployments/scripts/smoke-summoner.sh
Comment (cratelyn) [inline deployments/scripts/smoke-summoner.sh:18]: not strictly tied to the fix, but i found it to be very helpful to have a toggle to clear this state automatically, for the sake of a friendly|tight feedback loop during local development.

File: deployments/scripts/smoke-summoner.sh
Comment (cratelyn) [inline deployments/scripts/smoke-summoner.sh:1]: noting for future reference: i found when running this script locally that the `trap` commands herein weren't working as expected; `cometbft` and `pd` were sometimes left running, but i haven't tracked down what causes the `kill` call to fail. :raised_eyebrow:

File: flake.nix
Comment (cratelyn) [inline flake.nix:106]: cc, fellow nix flake user @plaidfinch: i found this was needed to put `cometbft` in the `$PATH` when running `nix develop`, in order to run the summonerd smoke test.

Comment (cratelyn) [reply flake.nix:106]: oh neat! can i suggest that we perform that flake maintenance in a separate issue, just to keep this narrowly focused upon repairing our failing CI?

Comment (cratelyn) [reply flake.nix:106]: filed #4369 to keep tabs on upgrading at some point :slightly_smiling_face:

File: deployments/scripts/smoke-summoner.sh
Comment (cratelyn) [reply deployments/scripts/smoke-summoner.sh:1]: filed #4370 to track that debugging effort. marked it as `_P-low` because it ties into development ergonomics. i'll mark this as resolved.

File: flake.nix
Comment (cratelyn) [reply flake.nix:106]: thanks! likewise, wanted to let you know in case that affects any possible validator infra. marking this as resolved now that you've seen it :slightly_smiling_face:

File: crates/bin/pcli/src/command/ceremony.rs
Comment (cratelyn) [inline crates/bin/pcli/src/command/ceremony.rs:61]: this is the fix :adhesive_bandage: 

everything else in this branch is tweaks to facilitate debugging, and confirming this _does_ in fact fix the issue.

**Review by hdevalence — APPROVED** [review]

File: deployments/scripts/smoke-summoner.sh
Comment (conorsch) [inline deployments/scripts/smoke-summoner.sh:70]: Thank you for the super detailed commit messages. :pray:

File: deployments/scripts/smoke-summoner.sh
Comment (conorsch) [reply deployments/scripts/smoke-summoner.sh:18]: Big ups. In general, I'd find it more ergonomic for the test suites to use tmpdirs across the board, but iirc pd actually requires a non-existent dir right now. Something to smooth over when next we're editing the pd cli tooling.

**Review by conorsch — APPROVED** [review]
Lovely, thanks for taking the time to show the work. Leaving unmerged, so you can drop/squash at your leisure.

File: deployments/scripts/smoke-summoner.sh
Comment (conorsch) [inline deployments/scripts/smoke-summoner.sh:126]: Hmm, I left a summoner-smoke running locally, and the pd process never exited. AIUI, each `trap` declaration clobbers the previous one, so I think what we really want is to restate the trap by appending pids to it. That way an exit/interrupt will still clean up the older processes. Not important to massage, given that this script only runs in CI, but mentioning it in case you have another round of clean up before merge.

**Pull request merged**
